File: app/src/main/java/com/yvrun/officeprocess/mvp/presenter/BasePresenter.java
package com.yvrun.officeprocess.mvp.presenter;

import com.yvrun.officeprocess.mvp.model.BaseModel;
import com.yvrun.officeprocess.mvp.view.IBaseView;

public abstract class BasePresenter<V extends IBaseView,M extends BaseModel>{

    protected M mModel;
    protected V mView;

    public BasePresenter(M model, V view) {
        mModel = model;
        mView = view;
    }
}

File: kotlinappother/src/main/java/com/chengzhen/kotlinappother/mvp/presenter/BasePresenter.kt
package com.chengzhen.kotlinappother.mvp.presenter

import com.chengzhen.kotlinappother.mvp.model.BaseModel
import com.chengzhen.kotlinappother.mvp.view.IBaseView

abstract class BasePresenter<V : IBaseView, M : BaseModel<*>>(protected var mModel: M, protected var mView: V)
